src/Authenticator/CookieAuthenticator.php

```diff
@@ -158,16 +158,19 @@ protected function _createPlainToken($identity): string
         $passwordField = $this->getConfig('fields.password');
 
         $salt = $this->getConfig('salt', '');
+
+        $value = $identity[$usernameField] . $identity[$passwordField];
+
+        if ($salt === false) {
+            return $value;
+        }
         if ($salt === true) {
             $salt = Security::getSalt();
-        } elseif ($salt === false) {
-            $salt = '';
-        } elseif (!is_string($salt) || $salt === '') {
+        }
+        if (!is_string($salt) || $salt === '') {
```

Comment:
This can still be `elseif`. Anyone setting Security salt to any empty string will have other problems already.

Reply:
Ah, that's why I added that, didn't know. Changed it.